syntax = "proto3";

package grpc;

message Nothing {}

message Id{
    string s = 1;
    int32 szx = 2;
    int32 szy = 3;
}

message GameInformation{
    int32 x = 1;
    int32 y = 2;
    string field = 3;
}

message Step{
    string id = 1;
    int32 move_x = 2;
    int32 move_y = 3;
}

service Game{
    rpc GetField (Id) returns (stream GameInformation);
    rpc MakeStep (Step) returns (Nothing);
}
